internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -277,3 +279,49 @@ func (ad *ActionDispatcher) scheduleRetry(ctx context.Context, action fleetapi.R
 		ad.log.Errorf("Unable to commit action retry (id %s) to fleet-server: %v", action.ID(), err)
 	}
 }
+
+func reportNextScheduledUpgrade(input []fleetapi.Action, detailsSetter details.Observer, log *logger.Logger) {
+	var nextUpgrade *fleetapi.ActionUpgrade
+	for _, action := range input {
+		sAction, ok := action.(fleetapi.ScheduledAction)
+		if !ok {
+			continue
+		}
+
+		uAction, ok := sAction.(*fleetapi.ActionUpgrade)
+		if !ok {
+			continue
+		}
+
+		start, err := uAction.StartTime()
+		if errors.Is(err, fleetapi.ErrNoStartTime) {
```

Comment:
in case error is a time not parsable then we will report scheduled for this unparsable value (we ignore errors when reporting). but we probably won't upgrade

Reply:
I changed the implementation in f31763a063 to check for _all_ errors returned by `uAction.StartTime()`, log them at an error level, and continue on so we don't consider such a `uAction` as the next scheduled upgrade action.